lib/zendesk2/help_center/translation.rb

```diff
@@ -0,0 +1,58 @@
+class Zendesk2::HelpCenter::Translation
+  include Zendesk2::Model
+
+  extend Zendesk2::Attributes
+
+  # @return [Integer] Automatically assigned when the translation is created
+  identity :id, type: :integer # ro:yes required:no
+
+  # @return [String] The API url of the translation
+  attribute :url, type: :string # ro:yes required:no
+  # @return [String] The url of the translation in Help Center
+  attribute :html_url, type: :string # ro:yes required:no
+  # @return [Integer]	The id of the item that has this translation
+  attribute :source_id, type: :integer # ro:yes required:no
+  # @return [String] The type of the item that has this translation. Can be Article, Section, or Category
+  attribute :source_type, type: :string # ro:yes required:no
+  # @return [String] The locale of the translation
+  attribute :locale, type: :string # ro:no required:yes
+  # @return [String] The title of the translation
+  attribute :title, type: :string # ro:no required:yes
+  # @return [String] The body of the translation. Empty by default
+  attribute :body, type: :string # ro:no required:no
+  # @return [Boolean] True if the translation is outdated; false otherwise. False by default
+  attribute :outdated, type: :boolean # ro:no required:no
+  # @return [Boolean] True if the translation is a draft; false otherwise. False by default
+  attribute :draft, type: :boolean # ro:no required:no
+
+  def destroy!
+    requires :locale, :source_id, :source_type
+
+    service.destroy_help_center_translation("translation" => Cistern::Hash.slice(self.attributes, :source_id, :source_type, :locale))
+  end
+
+  def save!
+    response = if new_record?
+                 requires :locale, :source_id, :source_type
+
+                 service.create_help_center_translation("translation" => self.attributes)
+               else
+                 requires :identity
+
+                 service.update_help_center_translation("translation" => self.attributes)
+               end
+
+    merge_attributes(response.body["translation"])
+  end
+
+  def source=(source)
+    if source.is_a?(Zendesk2::HelpCenter::Article)
+      self.source_type = "Article"
+    elsif source.is_a?(Zendesk2::HelpCenter::Section)
+      self.source_type = "Section"
+    elsif source.is_a?(Zendesk2::HelpCenter::Category)
+      self.source_type = "Category"
```

Comment:
Not sure what you mean here. The `source_type` is an [attribute on the resource](https://developer.zendesk.com/rest_api/docs/help_center/translations#json-format), the `source=` method is simply a helper for when you're creating a translation record:

```
client.help_center_translations.create!(source: article, title: "Hello", body: "World!", locale: "en-us")
```
vs
```
client.help_center_translations.create!(source_type: "Article", source_id: article.id, title: "Hello", body: "World!", locale: "en-us")
```

or are you just talking about removing from the model

```
# @return [Integer]	The id of the item that has this translation
attribute :source_id, type: :integer # ro:yes required:no
# @return [String] The type of the item that has this translation. Can be Article, Section, or Category
attribute :source_type, type: :string # ro:yes required:no
```

and adding them when the module is included?